File: trunk/LinuxTuner/MountManager/Menu.py
from PyQt4 import Qt

class Menu(Qt.QMenu) :
	def __init__(self, icon = None, text = None, parent = None) :
		Qt.QMenu.__init__(self, parent)
		
		self.caption_action = Qt.QWidgetAction(self)
		self.caption_action.setEnabled(False)

		self.caption_action_frame = Qt.QFrame()
		try : # FIXME: Check Qt.QFrameStyledPanel in PyQt-4.2
				self.caption_action_frame.setFrameShape(Qt.QFrame.StyledPanel) # Liksys Fix for KDE4
		except :
				self.caption_action_frame.setFrameShape(Qt.QFrame.Box) # KDE4
		self.caption_action.setDefaultWidget(self.caption_action_frame)

		self.caption_action_frame_layout = Qt.QHBoxLayout()
		self.caption_action_frame_layout.setMargin(3)
		self.caption_action_frame_layout.setSpacing(3)
		self.caption_action_frame.setLayout(self.caption_action_frame_layout)

		self.caption_action_icon_label = Qt.QLabel()
		icon_size = self.style().pixelMetric(Qt.QStyle.PM_SmallIconSize)
		if icon != None:
			self.caption_action_icon_label.setPixmap(icon.pixmap(Qt.QSize(icon_size, icon_size)))
		self.caption_action_frame_layout.insertWidget(-1, self.caption_action_icon_label, 0)

		self.caption_action_label = Qt.QLabel()
		if text != None:
			self.caption_action_label.setText(text)
		self.caption_action_frame_layout.insertWidget(-1, self.caption_action_label, 20)

		font = self.caption_action_label.font()
		font.setBold(True)
		self.caption_action_label.setFont(font)

		self.addAction(self.caption_action)
		 		    
        ### Public ###

	def setCaption(self,text) :
		self.caption_action_label.setText(text)

File: trunk/lightlang-editor/src/Menu.cpp
#include <QtGui/QWidgetAction>
#include <QtGui/QFrame>
#include <QtGui/QHBoxLayout>
#include <QtGui/QFont>
#include <QtGui/QLabel>
#include <QtGui/QStyle>
#include <QtGui/QIcon>
#include "Menu.h"

Menu::Menu(bool useSpecialHeader) {
	// Init
	headerAction = 0;
	headerActionFrame = 0;
	headerLabel = 0;
	iconLabel = 0;
	// Create header action if useSpecialHeader == true
	if (useSpecialHeader) {
		headerAction = new QWidgetAction(this);
		headerAction->setEnabled(false);
	
		headerActionFrame = new QFrame;
		headerActionFrame->setFrameShape(QFrame::Box);
	
		headerAction->setDefaultWidget(headerActionFrame);
	
		QHBoxLayout *headerActionFrameLayout = new QHBoxLayout();
		headerActionFrameLayout->setMargin(3);
		headerActionFrameLayout->setSpacing(3);
		headerActionFrame->setLayout(headerActionFrameLayout);

		iconLabel = new QLabel;
		headerLabel = new QLabel;
		
		iconSize = style()->pixelMetric(QStyle::PM_SmallIconSize);
		headerActionFrameLayout->insertWidget(-1,iconLabel,0);
		headerActionFrameLayout->insertWidget(-1,headerLabel,20);
		QFont font = headerLabel->font();
		font.setBold(true);
		headerLabel->setFont(font);
	
	
		addAction(headerAction);
	}
}

Menu::~Menu() {
	if (headerAction != 0) {
		delete headerLabel;
		delete iconLabel;
		delete headerActionFrame;
		delete headerAction;
	}
}

void Menu::setHeaderText(const QString &headerText) {
	if (headerLabel != 0)
		headerLabel->setText(headerText);
}

void Menu::setHeaderIcon(const QIcon &icon) {
	if (iconLabel != 0)
		iconLabel->setPixmap(icon.pixmap(iconSize,iconSize));
}